src/groupByKey.ts

```diff
@@ -2,6 +2,18 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T>(items: T[], key: T): GroupsMap<T> {
+  const result: GroupsMap<T> = {};
+
+  items.forEach((item) => {
+    const keyValue = item[key];
+
+    if (!result[keyValue as any]) {
+      result[keyValue as any] = [];
+    }
+
+    result[keyValue as any].push(item);
```

Comment:
Using 'keyValue as any' bypasses the type system which can lead to runtime errors. Instead, you should use a proper type assertion that aligns with the generic constraints.